projects/storefrontlib/src/lib/my-account/update-profile/components/update-profile-form.component.spec.ts

```diff
@@ -0,0 +1,167 @@
+  uid: 'xxx@xxx.xxx',
+};
+
+describe('UpdateProfileFormComponent', () => {
+  let component: UpdateProfileFormComponent;
+  let fixture: ComponentFixture<UpdateProfileFormComponent>;
+  let el: DebugElement;
+  let form: DebugElement;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [ReactiveFormsModule],
+      declarations: [UpdateProfileFormComponent],
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(UpdateProfileFormComponent);
+    component = fixture.componentInstance;
+    el = fixture.debugElement;
+    form = fixture.debugElement.query(By.css('form'));
+
+    fixture.detectChanges();
+    component.user = mockUser;
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  describe('isNotValid', () => {
+    it('should return true if NOT valid', () => {
+      component.ngOnInit();
+
+      expect(component.isNotValid('firstName')).toEqual(false);
+    });
+
+    it(`should return false if the form was not submitted even though it's invalid`, () => {
+      const invalidUser: User = {
+        ...mockUser,
+        firstName: '',
+      };
+      component.user = invalidUser;
+      component['submitClicked'] = false;
+
+      component.ngOnInit();
+      component.form.get('firstName').markAsDirty();
+
+      expect(component.isNotValid('firstName')).toEqual(false);
+    });
+
+    it('should return false if valid', () => {
+      const invalidUser: User = {
+        ...mockUser,
+        firstName: '',
+      };
+      component.user = invalidUser;
+      component['submitClicked'] = true;
+
+      component.ngOnInit();
+      component.form.get('firstName').markAsDirty();
+
+      expect(component.isNotValid('firstName')).toEqual(true);
+    });
+  });
+
+  describe('onSubmit', () => {
+    it('should call onSubmit() on submit event', () => {
+      spyOn(component, 'onSubmit').and.stub();
+      form.triggerEventHandler('ngSubmit', null);
+      expect(component.onSubmit).toHaveBeenCalled();
+    });
+
+    it('should NOT emit submited event if the form is not valid', () => {
+      spyOn(component.submited, 'emit').and.stub();
+
+      const invalidUser: User = {
+        ...mockUser,
+        firstName: '',
+      };
+      component.user = invalidUser;
+      component.ngOnInit();
+
+      component.onSubmit();
+      expect(component.submited.emit).not.toHaveBeenCalled();
+    });
+
+    it('should emit submited event', () => {
+      spyOn(component.submited, 'emit').and.stub();
+      component.user = mockUser;
+      component.ngOnInit();
+
+      component.onSubmit();
+      expect(component.submited.emit).toHaveBeenCalled();
+    });
+  });
+
+  describe('onSubmit', () => {
+    it('should call onCancel() on click event', () => {
+      spyOn(component, 'onCancel').and.stub();
+      const cancelBtn = el.query(By.css('button[type="button"]'));
+      cancelBtn.triggerEventHandler('click', null);
```

Comment:
Same comment as above about clicking on the button via the html element instead.